### ISSUE TITLE
`enum_variant_names` too trigger-happy

### ISSUE THREAD

**Issue by couchand** [rust-lang/rust-clippy#10599]
### Summary

I have an enum representing possible errors from deserialization:

```rust
pub enum PromotionError {
    NoPromotionId,
    InvalidPromotionId(String),
    InvalidProductId(String),
}
```

clippy produces this: `warning: all variants have the same postfix: Id` and "helpfully" suggests:  `help: remove the postfixes and use full paths to the variants instead of glob imports`

Maybe it's just me but the suggestion here seems completely irrelevant.  I'm not even sure what the lint is *trying* to catch: what's the stylistic concern here?

### Lint Name

enum_variant_names

### Reproducer

I tried this code:

```rust
pub enum PromotionError {
    NoPromotionId,
    InvalidPromotionId(String),
    InvalidProductId(String),
}
```

I saw this happen:

```
[<output>](warning: all variants have the same postfix: Id)
```

I expected to see this happen:

No error.

### Version

```text
rustc 1.68.1 (8460ca823 2023-03-20)
binary: rustc
commit-hash: 8460ca823e8367a30dda430efda790588b8c84d3
commit-date: 2023-03-20
host: x86_64-unknown-linux-gnu
release: 1.68.1
LLVM version: 15.0.6
```

**Comment by bonsairobo** [comment]
Yea I hit this as well and agree that the current implementation is annoying in some cases. The stated reason for the lint is to avoid post-fixing variants with the **enum type name**, so that's what the lint should check for.

**Comment by 7sDream** [comment]
This [code section](https://github.com/rust-lang/rust-clippy/blob/76a75bf145bbdfb688354d6f9bdebf3fc56af936/clippy_lints/src/item_name_repetitions.rs#L333-L379) does the check of all variant in a enum has a common prefix/postfix.

It makes sense that variants should not repeat enum names, but I don't see anything wrong with enums having a common prefix/postfix that is independent of the enum name.

At least there needs an configure option.

**Comment by spearman** [comment]
It would be nice if it actually checked for variants that repeated the enum type name instead of just any repeated string, which I don't find useful.
